### PR TITLE
Add GitHub action to automatically generate k8s yamls for ethereum 2.0 clients.

### PR DESCRIPTION
**Pull request by LaContra**
- Remove all yamls (hostPath and NFS) for all eth2 clietns.

### CODE REVIEW

File: lighthouse/host-path/beacon-deployment.yaml [hunk 1]
```diff
@@ -46,6 +48,6 @@ spec:
           path: /data/lighthouse/beacon
       securityContext:
         runAsUser: 1001
-        runAsGroup: 2000
+        runAsGroup: 2000
```
Comment (LaContra) [inline lighthouse/host-path/beacon-deployment.yaml:51]: There's an extra space. We can fix this format issue along with the security context value structure adjustment.

File: .github/workflows/helm-yaml-generation.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+          done
+
+      - name: Rearrange the locations for yaml files.
+        run: |
+          IFS="," read -ra CLIENTS <<< "$CLIENTS"
+          for client in "${CLIENTS[@]}"
+            do
+            for tempdir in ./$client/temp/*
+              do
+              dir=./$client/"${tempdir##*/}"             # Remove path.
```
Comment (chingyi-lin) [inline .github/workflows/helm-yaml-generation.yml:38]:
```suggestion
              dir="./{$client}/${tempdir##*/}"             # Remove path.
```

File: .github/workflows/helm-yaml-generation.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+              helm template ./$client/helm -f $fullfile --output-dir ./$client/temp/$foldername
+            done
+          done
+
+      - name: Rearrange the locations for yaml files.
+        run: |
+          IFS="," read -ra CLIENTS <<< "$CLIENTS"
+          for client in "${CLIENTS[@]}"
+            do
+            for tempdir in ./$client/temp/*
```
Comment (chingyi-lin) [inline .github/workflows/helm-yaml-generation.yml:36]:
```suggestion
            for tempdir in ./{$client}/temp/*
```

File: .github/workflows/helm-yaml-generation.yml [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+              do
+              dir=./$client/"${tempdir##*/}"             # Remove path.
+              if [ ! -d "$dir" ]; then mkdir $dir; fi  # Create yaml folder if it doesn't exist.
+              mv $tempdir/eth2-$client/templates/* $dir  # Move yamls to the correct folder.
+            done
+            rm -rf ./$client/temp
```
Comment (chingyi-lin) [inline .github/workflows/helm-yaml-generation.yml:42]:
```suggestion
            rm -rf ./{$client}/temp
```

File: .github/workflows/helm-yaml-generation.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    steps:
+      - name: Checkout target branch
+        uses: actions/checkout@v2 # https://github.com/actions/checkout
+
+      - name: Generate yamls for Ethereum 2.0 clients
+        run: |
+          IFS="," read -ra CLIENTS <<< "$CLIENTS"
+          for client in "${CLIENTS[@]}"
+            do
+            for fullfile in ./$client/helm/values-files/*
```
Comment (chingyi-lin) [inline .github/workflows/helm-yaml-generation.yml]:
```suggestion
            for fullfile in ./${client}/helm/values-files/* 
```

File: .github/workflows/helm-yaml-generation.yml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        run: |
+          IFS="," read -ra CLIENTS <<< "$CLIENTS"
+          for client in "${CLIENTS[@]}"
+            do
+            for fullfile in ./$client/helm/values-files/* 
+              do
+              filename="${fullfile##*/}"               # Remove path.
+              foldername="${filename%.*}"              # Remove file extension.
+              foldername="${foldername##*values-}"     # Remove 'values-' string.
+              helm template ./$client/helm -f $fullfile --output-dir ./$client/temp/$foldername
```
Comment (chingyi-lin) [inline .github/workflows/helm-yaml-generation.yml:27]:
```suggestion
              helm template ./{$client}/helm -f $fullfile --output-dir ./{$client}/temp/$foldername
```

File: .github/workflows/helm-yaml-generation.yml [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+      - name: Rearrange the locations for yaml files.
+        run: |
+          IFS="," read -ra CLIENTS <<< "$CLIENTS"
+          for client in "${CLIENTS[@]}"
+            do
+            for tempdir in ./$client/temp/*
+              do
+              dir=./$client/"${tempdir##*/}"             # Remove path.
+              if [ ! -d "$dir" ]; then mkdir $dir; fi  # Create yaml folder if it doesn't exist.
+              mv $tempdir/eth2-$client/templates/* $dir  # Move yamls to the correct folder.
```
Comment (chingyi-lin) [inline .github/workflows/helm-yaml-generation.yml:40]:
```suggestion
              mv {$tempdir}/eth2-{$client}/templates/* $dir  # Move yamls to the correct folder.
```

**Review by chingyi-lin — APPROVED** [review]

**Pull request merged**